### PR TITLE
Let backend return possibly adjusted commit on write

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/git_backend.rs
Comment (ilyagr) [inline lib/src/git_backend.rs:856]: It might be nice to test that the commit is correct as well, especially in the test where the timestamps change (so, that will test the following commit).

File: lib/src/git_backend.rs
Comment (ilyagr) [inline lib/src/git_backend.rs:540]: You probably considered this already, but it might be possible to simplify the code above a little using the fact that we now own `contents` and can modify it directly. Then, you wouldn't need to re-calculate the timestamps below.

This can go in a separate PR/commit, of course. Or is there some reason it doesn't work?

**Review by ilyagr — APPROVED** [review]

File: lib/src/git_backend.rs
Comment (martinvonz) [reply lib/src/git_backend.rs]: Good idea. See how this looks.

File: lib/src/git_backend.rs
Comment (martinvonz) [reply lib/src/git_backend.rs]: Yes, I did consider that but it wasn't obvious to me how to do it in a simpler way. We still need to recalculate the committer timestamp when we adjust it for collisions, right?

File: lib/src/git_backend.rs
Comment (ilyagr) [reply lib/src/git_backend.rs]: Another thing I would check there is that the commit id returned actually matches the commit returned. That would test that the timestamp manipulations you do are correct.

If this is less trivial than it seemed to me at first glance, we can skip it or postpone it.

File: lib/src/git_backend.rs
Comment (ilyagr) [reply lib/src/git_backend.rs]: I'm still thinking about this. Feel free to merge this before I try it; if I figure out some minor simplification I can always do another PR.

My minor annoyance is that we seem to be converting seconds to committer objects one too many times.

File: lib/src/git_backend.rs
Comment (martinvonz) [reply lib/src/git_backend.rs]: Ah, that's a good idea. Done. I also had to adjust the test case a bit because it ended up using negative timestamps, which libgit2 doesn't seem to preserve.

File: lib/src/git_backend.rs
Comment (martinvonz) [reply lib/src/git_backend.rs]: Sure, I'll leave that for you to look into if you like. Thanks.

**Pull request merged**
